web/pages/review.tsx

```diff
@@ -6,38 +6,39 @@ import { getFlashcards } from "@src/getFlashcards";
 import { useAuth } from "react-oidc-context";
 import { useQuery } from "@tanstack/react-query";
 import { getRequest } from "@src/apiRequest";
-
-const manualFlashcards = [
-    {
-        id: 0,
-        term: "Lorem ipsum crimble crumble that's just how the cookie crumbles",
-        definition:
-            "world! I'm going to talk about my feelings for a few minutes. Lately everything has",
-    },
-    {
-        id: 22,
-        term: "When",
-        definition:
-            "They ask you how you are and you just have to say that you're fine. But you're NOT really fine, you just can't get into it because they would never understand.",
-    },
-];
-
-const fillerFlashcards: Flashcard[] = [...new Array(10).keys()].map((i) => ({
-    id: i + 1,
-    term: "Hello",
-    definition: "world!",
-}));
-
-const flashcards = [...manualFlashcards, ...fillerFlashcards];
+import FolderListView from "@components/FolderListView";
+import { useCallback, useEffect, useState } from "react";
 
 const Review = () => {
     const auth = useAuth();
+    const [accessedFlashcards, setAccessedFlashcards] = useState<Flashcard[]>(
+        []
+    );
+
     // expand the index section when numbers exceed double digits (rel. to font size)
-    const fidLength = Math.max(...flashcards.map((f) => f.id)).toString()
-        .length;
-    const width = Math.max(6, fidLength * 2 + 2);
+    // TODO fix that this is called for every flashcard without using another state
+    function calcMaxWidth() {
+        const fidLength = Math.max(
+            ...accessedFlashcards.map((f) => f.id)
+        ).toString().length;
+        return Math.max(6, fidLength * 2 + 2);
+    }
+
+    const onSelectedFoldersChange = useCallback(
+        async (folderIds: number[]) => {
+            let flashcards: Flashcard[] = [];
+            if (!auth.user?.id_token) return flashcards;
+            for (const fId of folderIds) {
+                flashcards = flashcards.concat(
```

Comment:
Could use spread here? Not sure what's better?

Reply:
Turns out concat is more performant as it does not iterate over all the elements, and has array specific optimisations